### PR TITLE
chore: Nova4 additions

### LINKED ISSUE #10415: Better computation of NOVA 4 in German
[From the forum](https://forum.openfoodfacts.org/t/please-add-to-the-nova-4-list/579):

> At least for German products many which belong to the NOVA group 4 are missed and are only NOVA 3.

> * “modifizierte Stärke” = “modified starch” = chemically or physically treated plant (wheat, corn) starch to modify its properties
> * “Glukosesirup” = “glucose syrup” = plant (wheat, corn) starch treated with enzymes (amylases and amyloclucosidases) to split it in shorter chains up to glucose
> * “Glucose-Fructose-Sirup” = “glucose-fructose-syrup”, also known as “high fructose corn syrup” (HFCS), from enzyme treated starch as before, “Glukose” (glucose) and “Fruktose” (fuctose) in products are in many cases produced in this way, and should be also added
> * “*extrakt” e.g “Gewürzextrakt” = “*extract” e.g. “spice extract” = “100% natural” raw materials, from whatever sources, which has been fractional extracted and optionally mixed to form the declared extract, used as “natural” aroma in most cases
> * “*erzeugnis” e.g. “Milcheiweißerzeugnis” = catch-all declaration for all kinds or industrial proccessed “100% natural” raw substances
> * “whey protein concentrate” or “WPC” = size-fractionated whey protein as waste from cheese production, filtered by size by pressing it through menbranes

### PR DESCRIPTION
**Pull request by aleene**
### What
Made most suggested changes. Did not do `extract`. We need to have more details.

- Fixes #10415

### CODE REVIEW

File: taxonomies/food/ingredients.txt [hunk 1]
```diff
@@ -2113,6 +2113,7 @@ ko:발효유
 lv:Skābpiena produkti
 ru:Кисломолочные продукты
 uk:Кисломолочний продукт
+nova:en:4
```
Comment (stephanegigandet) [inline taxonomies/food/ingredients.txt:2116]: I don't think "fermented milk products" should be NOVA 4. e.g. plain yogurt is NOVA 1.

File: taxonomies/food/ingredients.txt [hunk 2]
```diff
@@ -4799,6 +4801,7 @@ hr:blagi punomasni jogurt
 
 <en:yogurt
 de:Joghurterzeugnis
+nova:en:4
```
Comment (stephanegigandet) [inline taxonomies/food/ingredients.txt:4804]: "Yogurt product" seems to be yogurt plus things like starch and ferments, it should not be NOVA 4.
e.g. "Joghurterzeugnis (9%) (fettarme Milch, Stärke, Joghurtkulturen)"

File: taxonomies/food/ingredients.txt [hunk 3]
```diff
@@ -5172,6 +5175,7 @@ it:derivati del siero di latte, la preparazione del siero di latte
 nl:Wei
 pl:produkt serwatkowy
 sv:vassleprodukt
+nova:en:4
```
Comment (stephanegigandet) [inline taxonomies/food/ingredients.txt:5178]: We should put "whey" directly in nova 4:
"Substances only found in ultra-processed products include some directly extracted from
foods, such as casein, lactose, whey, and gluten, and some derived from further processing
of food constituents, such as hydrogenated or interesterified oils, hydrolysed proteins, soy
protein isolate, maltodextrin, invert sugar and high fructose corn syrup. Classes of additive
only found in ultra-processed products include dyes and other colours, colour stabilisers,
flavours, flavour enhancers, non-sugar sweeteners, and processing aids such as
carbonating, firming, bulking and anti-bulking, de-foaming, anti-caking and glazing agents,
emulsifiers, sequestrants and humectants."
https://archive.wphna.org/wp-content/uploads/2016/01/WN-2016-7-1-3-28-38-Monteiro-Cannon-Levy-et-al-NOVA.pdf

File: taxonomies/food/ingredients.txt [hunk 4]
```diff
@@ -90846,6 +90854,7 @@ nl:sojabereiding
 pt:preparado à base de soja
 sv:sojabönsberedning
 allergens:en: en:soybeans
+nova:en:4
```
Comment (stephanegigandet) [inline taxonomies/food/ingredients.txt:90857]: soy preparation could be anything, it should not be NOVA 4.

Comment (aleene) [reply taxonomies/food/ingredients.txt:90857]: so we can not set anything. Or do we go for the worst case?

File: taxonomies/food/ingredients.txt (hunk 3 above)
Comment (aleene) [reply taxonomies/food/ingredients.txt:5178]: ok done

File: taxonomies/food/ingredients.txt (hunk 1 above)
Comment (aleene) [reply taxonomies/food/ingredients.txt:2116]: so should be set by children

Comment (aleene) [reply taxonomies/food/ingredients.txt:2116]: removed

File: taxonomies/food/ingredients.txt (hunk 2 above)
Comment (aleene) [reply taxonomies/food/ingredients.txt:4804]: In fact we do not know what happend to it. It is not simple yoghurt. If you look only at the ingredients you might miss hidden processing. What do we do with that? What is said about that?

Comment (aleene) [reply taxonomies/food/ingredients.txt:4804]: removed for now

File: taxonomies/food/ingredients.txt (hunk 4 above)
Comment (stephanegigandet) [reply taxonomies/food/ingredients.txt]: we don't go for the worse case. In most cases, we should have sub ingredients for things like "XYZ preparation", and those can trigger NOVA 3 or 4.

**Review by stephanegigandet — APPROVED** [review]
Thank you!

**Pull request merged**